pkg/controller/servicemesh/controlplane/readiness.go

```diff
@@ -209,36 +221,76 @@ func (r *controlPlaneInstanceReconciler) isCNIReady(ctx context.Context) (bool,
 	return true, nil
 }
 
-func (r *controlPlaneInstanceReconciler) calculateReadinessForType(ctx context.Context, list runtime.Object, isReady isReadyFunc, readinessMap map[string]bool) error {
+func (r *controlPlaneInstanceReconciler) calculateReadinessForType(ctx context.Context, namespaces []string, list runtime.Object, isReady isReadyFunc, readinessMap map[string]bool) error {
 	log := common.LogFromContext(ctx)
 
-	meshNamespace := r.Instance.GetNamespace()
-	selector := map[string]string{common.OwnerKey: meshNamespace}
-	err := r.Client.List(ctx, list, client.InNamespace(meshNamespace), client.MatchingLabels(selector))
-	if err != nil {
-		return err
-	}
-	items, err := meta.ExtractList(list)
-	if err != nil {
-		return err
-	}
-
-	for _, obj := range items {
-		metaObject, err := meta.Accessor(obj)
+	selector := map[string]string{common.OwnerKey: r.Instance.GetNamespace()}
+	for _, ns := range namespaces {
+		err := r.Client.List(ctx, list, client.InNamespace(ns), client.MatchingLabels(selector))
 		if err != nil {
 			return err
 		}
-		if component, ok := metaObject.GetLabels()[common.KubernetesAppComponentKey]; ok {
-			ready, exists := readinessMap[component]
-			readinessMap[component] = (ready || !exists) && isReady(obj)
-		} else {
-			// how do we have an owned resource with no component label?
-			log.Error(nil, "skipping resource for readiness check: resource has no component label", obj.GetObjectKind(), metaObject.GetName())
+		items, err := meta.ExtractList(list)
+		if err != nil {
+			return err
+		}
+
+		for _, obj := range items {
+			log.V(3).Info("Readiness check found object", "object", obj)
+			metaObject, err := meta.Accessor(obj)
+			if err != nil {
+				return err
+			}
+			if component, ok := metaObject.GetLabels()[common.KubernetesAppComponentKey]; ok {
+				ready, exists := readinessMap[component]
+				readinessMap[component] = (ready || !exists) && isReady(obj)
+			} else {
+				// how do we have an owned resource with no component label?
+				log.Error(nil, "skipping resource for readiness check: resource has no component label", obj.GetObjectKind(), metaObject.GetName())
```

Comment:
This should be an error or just a log?

Reply:
Good catch. This should be logged as V(3).Info, as it's definitely not an error that whoever is overseeing the operator needs to know about, since it can be caused by the user creating the resource.